### ISSUE TITLE
Getting upgrade your account message when selecting billing plan in development store

### ISSUE THREAD

**Issue by jithinjoyj4** [osiset/laravel-shopify#648]
*For bug reporting only! If you're posting a feature request or discussion, please ignore.*

## Expected Behavior

Ability to select billing plan for development store
## Current Behavior

Getting upgrade your account message when selecting billing plan with test=0 in development store as shown in screenshot
## Failure Information

Not able to select billing plan for development store, if test = 1 it works but the plan remain test and does not charge after the store transferred from development to real, thats why iam trying to use plan with test=0 in development store. if iam missing anything please help. any suggestion would be really helpful, Thanks

### Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Package Version: 13.1
* Laravel Version: 8.0
* PHP Version: vX.Y.Z
* Using a toolset (Docker, Laradock, Vagrant, etc.):aws server
![image_2020_12_24T14_26_43_434Z](https://user-images.githubusercontent.com/57803136/103095059-4462ec00-4625-11eb-9b3b-af5cebe6db2b.png)

### Failure Logs

Please include any relevant log snippets or files here.

**Comment by ncpope** [comment]
You have to create a new charge that is test=0

It doesn't matter if the store is in development or is on a paid Shopify plan. Test charges will not change on their own. You need to have a shop admin approve the charge

**Comment by arvesolland** [comment]
Actually the charge needs to have test=1 - that way it's just a test charge

**Comment by ncpope** [comment]
@arvesolland I read it as they said they wanted to convert a test charge to a charge that will actually bill the store

> but the plan remain test and does not charge after the store transferred from development to real

@jithinjoyj4 If that's not the case can you clarify your issue?

**Comment by jithinjoyj4** [comment]
@ncpope yes thats the case, development stores should be able to choose plans and after it transferred to client it should charge them as a partner friendly app. I know its a bit complicated since development store doesn't have billing option, currently iam planning to add a separate test plan for development store and update to real plan after it transferred, unless if there is any other way to do it with just one plan

**Comment by osiset** [comment]
I dont see this an issue with this repo, is the issue resolved? You can not bill a dev store, thus the screen you get.

**Comment by ncpope** [comment]
@jithinjoyj4 if that's the case then you will just need to create a new charge. There is no way to do it automatically without sending the merchant through the standard billing approval process.

This is a limitation imposed by shopify and this package and all other shopify app packages will not be able to do this for you automatically.

I would suggest that if you want to do this you can check the name of the merchants plan. If it is development or affiliate then keep it as  a test charge, otherwise redirect them to the billing screen.

Some pesudo code off the top of my head:

```php
$planName = $shop->api()->rest('GET', '/admin/shop.json')->body->shop->plan_name;

if ($planName === 'development') {
    return view('app.dashboard');
}

// create the new billing request
return redirect('/billing');
```

**Comment by jithinjoyj4** [comment]
Sorry for the delay. I've added separate test plan for development store and billing redirections done by checking shop plan name, Thank you for all the suggestions.
